PubNub/Network/Parsers/PNSubscribeParser.m

```diff
@@ -274,25 +270,28 @@ + (NSMutableDictionary *)messageFromData:(id)data
 
 + (NSMutableDictionary *)presenceFromData:(NSDictionary *)data {
     
+#warning ask about how this is affected
+    NSDictionary *payload = data[@"d"];
+    
     NSMutableDictionary *presence = [NSMutableDictionary new];
     
     // Processing common for all presence events data.
-    presence[@"presenceEvent"] = (data[@"action"]?: @"interval");
+    presence[@"presenceEvent"] = (payload[@"action"]?: @"interval");
```

Comment:
@jzucker2 Because data[kPNMessagePayloadDataKey] will be passed from calling method, there is no need to extract presence event data into separate variable and original code can be used with _data_ instead of _payload_.